app/src/main/java/com/techyos/oneclickonemail/ui/widgets/WidgetProvider.java

```diff
@@ -26,7 +27,7 @@
 
     @Override
     public void onUpdate(Context context, AppWidgetManager appWidgetManager, int[] appWidgetIds) {
-        WidgetEntryDao dao = new WidgetEntryDao(new DbHelper(context));
+        WidgetEntryDao dao = ((App)context.getApplicationContext()).getWidgetEntryDao();
```

Comment:
As the WidgetProvider is a broadcast receiver the applicationContext is created beforehand. (I always forget that for content providers this does not hold)